### PR TITLE
Fix the install script

### PR DESCRIPTION
**Pull request by LouDou**
- Allows ASDF_DOWNLOAD_PATH to be unset or empty
- Ensure the download actually happens
- Clean up temp dir after download and installation

### CODE REVIEW

File: bin/install [hunk 1]
```diff
@@ -12,26 +12,27 @@ my_mktemp () {
     local platform=""
     platform="$(get_platform)"
 
-    if [ "linux" = "$platform" ] ; then
-        tempdir=$(mktemp -d asdf-golang.XXXX)
-    else
-        tempdir=$(mktemp -dt asdf-golang.XXXX)
-    fi
-
+    tempdir=$(mktemp -dt asdf-golang.XXXX)
```
Comment (PierreBtz) [inline bin/install:15]: Unsure about this change, I couldn't test it as I'm on darwin.

**Review by PierreBtz — COMMENTED** [review]
I can confirm this indeed fixes an issue with the `ASDF_DOWNLOAD_PATH` that I also noticed after a recent upgrade of the plugin. I couldn't test the whole change though, see inline comment.

File: bin/install (hunk 1 above)
Comment (LouDou) [reply bin/install:15]: On Ubuntu 20.20
```$ mktemp --help
Usage: mktemp [OPTION]... [TEMPLATE]
Create a temporary file or directory, safely, and print its name.
TEMPLATE must contain at least 3 consecutive 'X's in last component.
If TEMPLATE is not specified, use tmp.XXXXXXXXXX, and --tmpdir is implied.
Files are created u+rw, and directories u+rwx, minus umask restrictions.

  -d, --directory     create a directory, not a file
  -u, --dry-run       do not create anything; merely print a name (unsafe)
  -q, --quiet         suppress diagnostics about file/dir-creation failure
      --suffix=SUFF   append SUFF to TEMPLATE; SUFF must not contain a slash.
                        This option is implied if TEMPLATE does not end in X
  -p DIR, --tmpdir[=DIR]  interpret TEMPLATE relative to DIR; if DIR is not
                        specified, use $TMPDIR if set, else /tmp.  With
                        this option, TEMPLATE must not be an absolute name;
                        unlike with -t, TEMPLATE may contain slashes, but
                        mktemp creates only the final component
  -t                  interpret TEMPLATE as a single file name component,
                        relative to a directory: $TMPDIR, if set; else the
                        directory specified via -p; else /tmp [deprecated]
      --help     display this help and exit
      --version  output version information and exit

GNU coreutils online help: <https://www.gnu.org/software/coreutils/>
Full documentation at: <https://www.gnu.org/software/coreutils/mktemp>
or available locally via: info '(coreutils) mktemp invocation'
```
It seems to be the same

**Review by aheritier — APPROVED** [review]

**Comment by kennyp** [comment]
Thanks for the PR!

A few questions...

- Why not use `true`/`false` instead of `"1"`/`"0"`?
- If I remember correctly, the reason for leaving of the `-t` on Linux had to do with some versions of `mktemp` not supporting it _and_ it not being required. Not to mention according to the man page the `-t` flag is deprecated on the coreutils version.

**Comment by kennyp** [comment]
Seems to work for me @LouDou! I'm going to go ahead and merge and we can bikeshed my questions later. :smile:

**Pull request merged**